### PR TITLE
Feat/write acls receive doc

### LINKED ISSUE #838: Integrate CollaboratorCanUpdate - 2
Integrate the call when Receiving the anchored document.

Criteria:
1. Tests good and bad
2. testworld

### PR DESCRIPTION
**Pull request by vedhavyas**
Fixes #838 

Also, removed creating identities in every test. this should improve our integration tests run time

Note: testworld tests that check for transition failures are disabled until #807 is done

### CODE REVIEW

File: testworld/document_consensus_test.go [hunk 1]
```diff
@@ -74,6 +74,10 @@ func addExternalCollaborator_withinHost(t *testing.T, documentType string) {
 	getDocumentAndCheck(bob.httpExpect, b, documentType, params)
 	nonExistingDocumentCheck(bob.httpExpect, c, documentType, params)
 
+	//// let c update the document and fail
+	//res = failedUpdateDocument(bob.httpExpect, c, documentType, http.StatusInternalServerError, docIdentifier, updatedDocumentPayload(documentType, []string{a, c}))
+	//assert.NotNil(t, res)
```
Comment (vedhavyas) [inline testworld/document_consensus_test.go]: We can enable this once we have the #807 is done

**Comment by vedhavyas** [comment]
I don't understand codecov. Test coverage should be more than 95 % :(

File: documents/documents_test/service_test.go [hunk 2]
```diff
@@ -340,9 +419,18 @@ func TestService_Exists(t *testing.T) {
 
 }
 
-func createCDWithEmbeddedInvoice(t *testing.T, ctx context.Context, skipSave bool) (documents.Model, coredocumentpb.CoreDocument) {
+func createCDWithEmbeddedInvoice(t *testing.T, ctx context.Context, collaborators []identity.DID, skipSave bool) (documents.Model, coredocumentpb.CoreDocument) {
```
Comment (charleenfei) [inline documents/documents_test/service_test.go]: maybe this function could be moved to the testingdocuments package

Comment (vedhavyas) [reply documents/documents_test/service_test.go]: That would create a dependency cycle AFAIK

**Pull request merged**